ec/ecresource/deploymentresource/schema_v0.go

```diff
@@ -530,6 +535,116 @@ func resourceSchemaV0() *schema.Resource {
+									Description: `Optional size type, defaults to "memory".`,
+									Default:     "memory",
+									Optional:    true,
+								},
+								"zone_count": {
+									Type:     schema.TypeInt,
+									Computed: true,
+									Optional: true,
+								},
+							},
+						},
+					},
+
+					"config": {
+						Type:             schema.TypeList,
+						Optional:         true,
+						MaxItems:         1,
+						DiffSuppressFunc: suppressMissingOptionalConfigurationBlock,
+						Description:      `Optionally define the IntegrationsServer configuration options for the IntegrationsServer Server`,
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								// IntegrationsServer System Settings
+								"debug_enabled": {
+									Type:        schema.TypeBool,
+									Description: `Optionally enable debug mode for IntegrationsServer servers - defaults to false`,
+									Optional:    true,
+									Default:     false,
+								},
+
+								"user_settings_json": {
+									Type:        schema.TypeString,
+									Description: `An arbitrary JSON object allowing (non-admin) cluster owners to set their parameters (only one of this and 'user_settings_yaml' is allowed), provided they are on the whitelist ('user_settings_whitelist') and not on the blacklist ('user_settings_blacklist'). (This field together with 'user_settings_override*' and 'system_settings' defines the total set of resource settings)`,
+									Optional:    true,
+								},
+								"user_settings_override_json": {
+									Type:        schema.TypeString,
+									Description: `An arbitrary JSON object allowing ECE admins owners to set clusters' parameters (only one of this and 'user_settings_override_yaml' is allowed), ie in addition to the documented 'system_settings'. (This field together with 'system_settings' and 'user_settings*' defines the total set of resource settings)`,
+									Optional:    true,
+								},
+								"user_settings_yaml": {
+									Type:        schema.TypeString,
+									Description: `An arbitrary YAML object allowing ECE admins owners to set clusters' parameters (only one of this and 'user_settings_override_json' is allowed), ie in addition to the documented 'system_settings'. (This field together with 'system_settings' and 'user_settings*' defines the total set of resource settings)`,
+									Optional:    true,
+								},
+								"user_settings_override_yaml": {
+									Type:        schema.TypeString,
+									Description: `An arbitrary YAML object allowing (non-admin) cluster owners to set their parameters (only one of this and 'user_settings_json' is allowed), provided they are on the whitelist ('user_settings_whitelist') and not on the blacklist ('user_settings_blacklist'). (These field together with 'user_settings_override*' and 'system_settings' defines the total set of resource settings)`,
+									Optional:    true,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
```

Comment:
This is not necessary since it didn't exist before. This function is only used to migrate from previous schema versions to new. It would be relevant if we were removing fields, changing defaults or similar.